chainer/training/updater.py

```diff
@@ -127,6 +127,9 @@ def __init__(self, iterator, optimizer, converter=convert.concat_examples,
             optimizer = {'main': optimizer}
         self._optimizers = optimizer
 
+        if device and device >= 0:
```

Comment:
This line does not work correctly if `device == 0`. I believe that this should be `if device is not None and device >= 0:`.